nimble/host/src/ble_hs_id.c

```diff
@@ -58,9 +58,21 @@ ble_hs_id_set_rnd(const uint8_t *rnd_addr)
 {
     uint8_t addr_type_byte;
     int rc;
+    uint16_t sum;
 
     ble_hs_lock();
 
+    /* Make sure all bits are neither one nor zero */
+     sum = 0;
+     for (rc = 0; rc < BLE_DEV_ADDR_LEN; ++rc) {
+         sum += rnd_addr[rc];
+     }
+
+     if ((sum == 0) || (sum == (6*255))) {
```

Comment:
that's counter-intuitive tbh, why not just check if any byte is other that 0x00 or 0xff in loop above?

Reply:
Well I just tried to keep the implementation similar to `ble_ll_is_valid_random_addr`.